### ISSUE TITLE
pact-jvm-master setup

### ISSUE THREAD

**Issue by nigel-millward** [DiUS/pact-jvm#504]
I am trying to setup the pact-jvm-master project, and when building the project I am getting errors. So far I have: 

- cloned the project from github - https://github.com/DiUS/pact-jvm
- open in intellij as existing project and imported using gradle
- Added scala as intellij plugin
- Added scala sdk to project within Project Structure | Global Libraries

When I go to Build | Build project, I am seeing 6 errros with Matchers.scala. Some examples of errors below:

1.pact-jvm-master\pact-jvm-matchers\src\main\scala\au\com\dius\pact\matchers\Matchers.scala
Error(76,37) not found: type MismatchFactory mismatchFn:MismatchFactory[Mismatch]):List[Mismatch] =  {
 
 def domatch[Mismatch](matchers: MatchingRules, category: String, path: Seq[String], expected: Any, actual: Any,
                        mismatchFn: MismatchFactory[Mismatch]) : List[Mismatch] = {
    val matcherDef = selectBestMatcher(matchers, category, path)
    asScalaBuffer(MatcherExecutorKt.domatch(matcherDef, path, expected, actual, mismatchFn)).toList
  }
  
2. pact-jvm-master\pact-jvm-matchers\src\main\scala\au\com\dius\pact\model\Matching.scala
Error:73,38) not found: type MismatchFactory, object HeaderMismatchFactory extends MismatchFactory[HeaderMismatch]{

object HeaderMismatchFactory extends MismatchFactory[HeaderMismatch] {
  import JavaConversions._
  def create(expected: Object, actual: Object, message: String, path: java.util.List[String]) = {
    HeaderMismatch(path.toList.last, expected.toString, actual.toString, Some(message))
  }
}

 Is there anything else I need to do with Scala so that the project builds?

**Comment by uglyog** [comment]
Remove the Scala SDK from the Global Libraries, IntelliJ will import all the required settings from the Gradle build file.

To build the project, use the gradle scripts in the root (gradlew for Linux and OSX, gradlew.bat for Windows) and run them from a command line terminal.

```console
$ ./gradle build
```

This will compile all the source files, generate the jar files and execute all the tests.

If you want to run it from within IntelliJ, use the Gradle Tool window and double click the build task (you'll need to find it in the tree under `pact-jvm | pact-jvm (root) | Tasks | build | build`.

**Comment by nigeltest** [comment]
Thanks for the help.  I have removed the Scala SDK from the global libraries and running the build through the command line with gradle. The build  fails on task ':pact-jvm-matchers_2.11:compileScala.' 

![gradlebuildexception](https://user-images.githubusercontent.com/12253614/28826591-882b09bc-76c2-11e7-8ca4-91b8baa177c0.png)

![exception_pactjvmmatchers211compilescala](https://user-images.githubusercontent.com/12253614/28826599-8b8019e0-76c2-11e7-8cd7-804430230acd.png)

**Comment by nigeltest** [comment]
think I worked it out. if I run build.gradle from specific project, eg pact-jvm-consumer-junit, or pat-jvm-provider-junit then the project compiles successfully. If i run gradle.build from root pact-jvm-master then I get scala compilation errors.

**Comment by uglyog** [comment]
Looks like Gradle is not compiling the source code in the correct order on your machine.

**Comment by uglyog** [comment]
Aah, your're running the Gradle installed on your machine, which is probably version 4.0.

Use the `gradlew.bat` wrapper script in the root of the project. It will use the correct version of Gradle.

**Comment by nigeltest** [comment]
Awesome, thanks. I was using gradle 4.02. but now using gradlew.bat and running tests from root pact-jvm-master from the command line.  I think the wrapper is using gradle 3.5

One thing I did notice is that the junit consumer tests were failing with the error: 'java.lang.NoClassDefFoundError: Could not initialize class au.com.dius.pact.model.RequestMatching$'

To stop this error I added the dependencies below into pact-jvm-master/build.gradle, as  there was possibly a dependency mismatch
        compile group: 'au.com.dius', name: 'pact-jvm-provider_2.11', version: '3.5.2'
        compile group: 'au.com.dius', name: 'pact-jvm-consumer_2.11', version: '3.5.2'
